static/app/components/profiling/BoundTooltip.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+import {mat3, vec2} from 'gl-matrix';
+
+import {FlamegraphTheme} from 'sentry/utils/profiling/flamegraph/FlamegraphTheme';
+import {getContext, measureText, Rect} from 'sentry/utils/profiling/gl/utils';
+
+const useCachedMeasure = (string: string): Rect => {
+  const cache = React.useRef<Record<string, Rect>>({});
+  const ctx = React.useMemo(() => {
+    const context = getContext(document.createElement('canvas'), '2d');
+
+    context.font = `12px ui-monospace, Menlo, Monaco, 'Cascadia Mono', 'Segoe UI Mono', 'Roboto Mono',
+    'Oxygen Mono', 'Ubuntu Monospace', 'Source Code Pro', 'Fira Mono', 'Droid Sans Mono',
+    'Courier New', monospace`;
+    return context;
+  }, []);
+
+  return React.useMemo(() => {
+    if (cache.current[string]) {
+      return cache.current[string];
+    }
+
+    if (!ctx) {
+      return Rect.Empty();
+    }
+
+    const measures = measureText(string, ctx);
+    cache.current[string] = measures;
+
+    return new Rect(0, 0, measures.width, measures.height);
+  }, [string, ctx]);
+};
+
+interface BoundTooltipProps {
+  bounds: Rect;
+  configToPhysicalSpace: mat3;
+  cursor: vec2 | null;
+  theme: FlamegraphTheme;
+  children?: React.ReactNode;
+}
+
+function BoundTooltip({
+  bounds,
+  configToPhysicalSpace,
+  cursor,
+  children,
+  theme,
+}: BoundTooltipProps): React.ReactElement | null {
+  const tooltipRef = React.useRef<HTMLDivElement>(null);
+  const tooltipRect = useCachedMeasure(tooltipRef.current?.textContent ?? '');
+
+  const physicalToLogicalSpace = React.useMemo(
+    () =>
+      mat3.fromScaling(
+        mat3.create(),
+        vec2.fromValues(1 / window.devicePixelRatio, 1 / window.devicePixelRatio)
+      ),
+    [window.devicePixelRatio]
+  );
```

Comment:
I don't think adding `window.devicePixelRatio` in the dependencies actually accomplishes what you want. Check it out [here](https://codesandbox.io/s/window-devicepixelratio-dependency-zj6lp?file=/src/App.js)

Reply:
Ah true, I will use this example - https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio